src/expr/impl/src/scalar/cast.rs

```diff
@@ -241,6 +241,17 @@ fn struct_cast(input: StructRef<'_>, ctx: &Context) -> Result<StructValue> {
     Ok(StructValue::new(fields))
 }
 
+/// Cast array with `source_elem_type` into array with `target_elem_type` by casting each element.
+#[function("cast(anymap) -> anymap", type_infer = "panic")]
+fn map_cast(map: MapRef<'_>, ctx: &Context) -> Result<MapValue> {
+    let new_ctx = Context {
+        arg_types: vec![ctx.arg_types[0].clone().as_map().clone().into_list()],
+        return_type: ctx.return_type.as_map().clone().into_list(),
+        variadic: ctx.variadic,
+    };
```

Comment:
The real reason was that I didn't fully understand what `list_cast` is doing, so I just chose to invoke it directly instead of imitating. 🤡

https://github.com/risingwavelabs/risingwave/blob/34e98292ba8fb83b7992f82576cb20594680a37e/src/expr/impl/src/scalar/cast.rs#L200-L205

Reply:
Well, `list_cast` also clones datatypes and allocates vec every time..